devsetup/ceph/deploy.sh

```
@@ -94,7 +94,23 @@ get_ceph_cli() {
 
 distribute_keys() {
     local ip="$1"
-    ssh-copy-id -o StrictHostKeyChecking=no -i "$DEFAULT_CEPH_PUB" root@"$ip"
+    local temp_known_hosts="$HOME/.ssh/ceph_known_hosts.tmp"
+
+    # Collect host keys from the newly deployed target node
+    echo "Collecting SSH host keys from $ip..."
+    ssh-keyscan -H "$ip" >> "$temp_known_hosts" 2>/dev/null
+
+    if [ $? -ne 0 ]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)